universal-login-commons/lib/core/utils/authorisation.ts

```diff
@@ -0,0 +1,24 @@
+import {utils} from 'ethers';
+
+import {CancelAuthorisationRequest} from '../models/authorisation';
+
+export const hashCancelAuthorisationRequest =
+  (cancelAuthorisationRequest: CancelAuthorisationRequest): string => {
+    const payloadString = JSON.stringify(cancelAuthorisationRequest);
+    const payloadBytes = utils.toUtf8Bytes(payloadString);
+    return utils.keccak256(payloadBytes);
+  };
+
+export const signCancelAuthorisationRequest =
+  (cancelAuthorisationRequest: CancelAuthorisationRequest, privateKey: string): utils.Signature => {
```

Comment:
utils.Signature is complex type, I would serialize it to hex string (which I think is what ethers do automatically by default). #stringsignature